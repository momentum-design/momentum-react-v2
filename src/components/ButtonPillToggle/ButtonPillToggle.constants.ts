import { SIZES } from '../ButtonPill/ButtonPill.constants';

const CLASS_PREFIX = 'md-button-pill-toggle';

const DEFAULTS = {
  SELECTED: false,
  OUTLINE: undefined,
  GHOST: true,
  DISABLED: false,
};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { CLASS_PREFIX, DEFAULTS, SIZES, STYLE };
