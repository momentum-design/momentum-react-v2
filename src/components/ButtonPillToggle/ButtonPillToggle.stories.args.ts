import { commonStyles, commonAriaButtonToggle } from '../../storybook/helper.stories.argtypes';
import { BUTTON_PILL_TOGGLE_CONSTANTS as CONSTANTS } from './';
import buttonPillArgTypes from '../ButtonPill/ButtonPill.stories.args';

const buttonPillToggleArgTypes = {
  ...buttonPillArgTypes,
  ...commonAriaButtonToggle,
};

delete buttonPillToggleArgTypes.color;

buttonPillToggleArgTypes.disabled.table.defaultValue.summary = CONSTANTS.DEFAULTS.DISABLED;
buttonPillToggleArgTypes.ghost.table.defaultValue.summary = CONSTANTS.DEFAULTS.GHOST;
buttonPillToggleArgTypes.outline.table.defaultValue.summary = CONSTANTS.DEFAULTS.OUTLINE;

export { buttonPillToggleArgTypes };

export default {
  ...commonStyles,
  ...buttonPillToggleArgTypes,
};
