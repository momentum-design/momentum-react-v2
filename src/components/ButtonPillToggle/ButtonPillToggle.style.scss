.md-button-pill-toggle-wrapper {
  &[data-selected='true'] {
    &[data-outline='true'],
    &[data-ghost='true'] {
      background-color: var(--mds-color-theme-button-secondary-active-normal) !important;
      color: var(--mds-color-theme-text-primary-normal) !important;

      &:hover,
      &.hover {
        background-color: var(--mds-color-theme-button-secondary-active-hover) !important;
        color: var(--mds-color-theme-text-primary-normal) !important;
      }

      &:active,
      &.active {
        background-color: var(--mds-color-theme-button-secondary-active-pressed) !important;
        color: var(--mds-color-theme-text-primary-normal) !important;
      }

      &[data-disabled='true'],
      &.disable {
        background-color: var(--mds-color-theme-button-secondary-active-normal) !important;
        color: var(--mds-color-theme-text-primary-disabled) !important;
      }
    }

    &[data-outline='true'] {
      border-color: var(--mds-color-theme-outline-button-normal) !important;

      &:hover,
      &.hover {
        border-color: var(--mds-color-theme-outline-button-normal) !important;
      }

      &:active,
      &.active {
        border-color: var(--mds-color-theme-outline-button-normal) !important;
      }

      &[data-disabled='true'],
      &.disable {
        border-color: var(--mds-color-theme-outline-primary-disabled) !important;
      }
    }
  }
}
