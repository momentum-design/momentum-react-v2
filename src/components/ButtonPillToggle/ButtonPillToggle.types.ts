import { ButtonPillProps } from '../ButtonPill';
import { AriaButtonProps, AriaToggleButtonProps } from '@react-types/button';

export interface Props
  extends Omit<ButtonPillProps, keyof AriaButtonProps>,
    AriaToggleButtonProps {}
