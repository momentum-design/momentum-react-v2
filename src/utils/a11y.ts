import { AriaAttributes, useEffect } from 'react';
import { RequireOneOf } from './types';

export type AriaLabelRequired = RequireOneOf<AriaAttributes, ['aria-label', 'aria-labelledby']>;

/**
 * Check if `aria-label` or `aria-labelledby` are defined and are truthy
 *
 * @param componentName The name of the component
 * @param props The object of props passed to the component
 */
export function useCheckAriaLabel(
  componentName: string,
  props: { 'aria-label'?: string; 'aria-labelledby'?: string }
): void {
  useEffect(() => {
    if (!props['aria-label'] && !props['aria-labelledby']) {
      console.warn(`MRV2: ${componentName} requires aria-label or aria-labelledby.`);
    }
  }, [componentName, props]);
}
