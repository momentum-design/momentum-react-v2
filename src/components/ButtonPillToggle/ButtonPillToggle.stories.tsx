import { MultiTemplateWithPseudoStates, Template } from '../../storybook/helper.stories.templates';
import { DocumentationPage } from '../../storybook/helper.stories.docs';
import StyleDocs from '../../storybook/docs.stories.style.mdx';

import ButtonPillToggle, { ButtonPillToggleProps } from './';
import argTypes from './ButtonPillToggle.stories.args';
import Documentation from './ButtonPillToggle.stories.docs.mdx';

export default {
  title: 'Momentum UI/ButtonPillToggle',
  component: ButtonPillToggle,
  parameters: {
    expanded: true,
    docs: {
      page: DocumentationPage(Documentation, StyleDocs),
    },
    args: {
      style: { margin: '0.5rem' },
    },
  },
};

const Example = Template<ButtonPillToggleProps>(ButtonPillToggle).bind({});

Example.argTypes = { ...argTypes };

Example.args = {
  children: 'Example text',
};

// Pseudostates will be correctly applied to ButtonPillToggle once ButtonPill
// is fixed to receive these pseudostates.

const Outline = MultiTemplateWithPseudoStates<ButtonPillToggleProps>(ButtonPillToggle).bind({});

Outline.argTypes = { ...argTypes };
delete Outline.argTypes.children;
delete Outline.argTypes.ghost;
delete Outline.argTypes.outline;
delete Outline.argTypes.isSelected;

Outline.args = {
  children: 'Example text',
};

Outline.parameters = {
  variants: [{ outline: true }, { outline: true, isSelected: true }],
};

const Ghost = MultiTemplateWithPseudoStates<ButtonPillToggleProps>(ButtonPillToggle).bind({});

Ghost.argTypes = { ...argTypes };
delete Ghost.argTypes.children;
delete Ghost.argTypes.ghost;
delete Ghost.argTypes.outline;
delete Ghost.argTypes.isSelected;

Ghost.args = {
  children: 'Example text',
};

Ghost.parameters = {
  variants: [{ ghost: true }, { ghost: true, isSelected: true }],
};

export { Example, Outline, Ghost };
