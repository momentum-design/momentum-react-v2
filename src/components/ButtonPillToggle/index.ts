import { default as ButtonPillToggle } from './ButtonPillToggle';
import * as CONSTANTS from './ButtonPillToggle.constants';
import { Props } from './ButtonPillToggle.types';

export { CONSTANTS as BUTTON_PILL_TOGGLE_CONSTANTS };

export type ButtonPillToggleProps = Props;

export default ButtonPillToggle;
