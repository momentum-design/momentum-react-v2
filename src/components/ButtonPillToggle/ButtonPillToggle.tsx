import React, { RefObject, forwardRef, useRef } from 'react';
import classnames from 'classnames';
import ButtonPill from '../ButtonPill';
import { useToggleState } from '@react-stately/toggle';

import { DEFAULTS, STYLE } from './ButtonPillToggle.constants';
import { DEFAULTS as BUTTON_PILL_DEFAULTS } from '../ButtonPill/ButtonPill.constants';
import { Props } from './ButtonPillToggle.types';
import './ButtonPillToggle.style.scss';
import { chain } from '@react-aria/utils';

const ButtonPillToggle = forwardRef((props: Props, providedRef: RefObject<HTMLButtonElement>) => {
  const {
    children,
    className,
    ghost = DEFAULTS.GHOST,
    outline = DEFAULTS.OUTLINE,
    disabled = DEFAULTS.DISABLED,
    size = BUTTON_PILL_DEFAULTS.SIZE,
    onPress,
    ...otherProps
  } = props;

  const internalRef = useRef();
  const ref = providedRef || internalRef;

  const state = useToggleState(props);

  if (ghost === false) {
    console.warn('MRV2: Momentum does not support non-ghost ButtonPillToggle.');
  }

  return (
    <ButtonPill
      className={classnames(STYLE.wrapper, className)}
      outline={outline}
      ghost={ghost}
      disabled={disabled}
      size={size}
      data-selected={state.isSelected || DEFAULTS.SELECTED}
      onPress={chain(state.toggle, onPress)}
      aria-pressed={state.isSelected}
      {...otherProps}
      ref={ref}
    >
      {children}
    </ButtonPill>
  );
});

ButtonPillToggle.displayName = 'ButtonPillToggle';

export default ButtonPillToggle;
