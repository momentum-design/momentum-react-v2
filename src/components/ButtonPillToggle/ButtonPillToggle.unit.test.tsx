import React from 'react';
import { mount } from 'enzyme';

import ButtonPillToggle, { BUTTON_PILL_TOGGLE_CONSTANTS as CONSTANTS } from './';
import { triggerPress } from '../../../test/utils';

describe('<ButtonPillToggle />', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      const container = mount(<ButtonPillToggle />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with isSelected being true', () => {
      expect.assertions(1);

      const isSelected = true;

      const container = mount(<ButtonPillToggle isSelected={isSelected} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with isSelected being false', () => {
      expect.assertions(1);

      const isSelected = false;

      const container = mount(<ButtonPillToggle isSelected={isSelected} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with ghost being true', () => {
      expect.assertions(1);

      const ghost = true;

      const container = mount(<ButtonPillToggle ghost={ghost} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with ghost being false', () => {
      expect.assertions(1);

      const ghost = false;

      const container = mount(<ButtonPillToggle ghost={ghost} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with outline being true', () => {
      expect.assertions(1);

      const outline = true;

      const container = mount(<ButtonPillToggle outline={outline} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with outline being false', () => {
      expect.assertions(1);

      const outline = false;

      const container = mount(<ButtonPillToggle outline={outline} />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', () => {
      expect.assertions(1);

      const element = mount(<ButtonPillToggle />)
        .find(ButtonPillToggle)
        .getDOMNode();

      expect(element.classList.contains(CONSTANTS.STYLE.wrapper)).toBe(true);
    });

    it('should have provided outline when outline is provided', () => {
      expect.assertions(1);

      const outline = true;

      const element = mount(<ButtonPillToggle outline={outline} />)
        .find(ButtonPillToggle)
        .getDOMNode();

      expect(element.getAttribute('data-outline')).toBe(`${outline}`);
    });

    it('should have provided ghost when ghost is provided', () => {
      expect.assertions(1);

      const ghost = true;

      const element = mount(<ButtonPillToggle ghost={ghost} />)
        .find(ButtonPillToggle)
        .getDOMNode();

      expect(element.getAttribute('data-ghost')).toBe(`${ghost}`);
    });

    it('should have provided isSelected when isSelected is provided', () => {
      expect.assertions(1);

      const isSelected = true;

      const element = mount(<ButtonPillToggle isSelected={isSelected} />)
        .find(ButtonPillToggle)
        .getDOMNode();

      expect(element.getAttribute('data-selected')).toBe(`${isSelected}`);
    });
  });

  describe('actions', () => {
    it('onChange callback is called correctly when provided', () => {
      expect.assertions(1);

      const mockCallback = jest.fn();

      const initialIsSelected = false;

      const component = mount(<ButtonPillToggle onChange={mockCallback} />).find(ButtonPillToggle);

      component.props().onChange(initialIsSelected);

      expect(mockCallback).toBeCalledTimes(1);
    });

    it('should handle mouse press events', () => {
      expect.assertions(4);

      const mockCallback = jest.fn();

      const component = mount(<ButtonPillToggle onChange={mockCallback} />).find(ButtonPillToggle);

      triggerPress(component);

      expect(mockCallback).toHaveBeenCalledWith(true);

      triggerPress(component);

      expect(mockCallback).toHaveBeenCalledWith(false);

      triggerPress(component);

      expect(mockCallback).toHaveBeenCalledWith(true);

      triggerPress(component);

      expect(mockCallback).toHaveBeenCalledWith(false);
    });

    it('should handle mouse press event when disabled', () => {
      expect.assertions(1);

      const mockCallback = jest.fn();

      const component = mount(<ButtonPillToggle onChange={mockCallback} disabled />).find(
        ButtonPillToggle
      );

      triggerPress(component);

      expect(mockCallback).toBeCalledTimes(0);
    });
  });
});
