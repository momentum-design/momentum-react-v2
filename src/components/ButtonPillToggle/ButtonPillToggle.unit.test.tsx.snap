// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonPillToggle /> snapshot should match snapshot 1`] = `
<ButtonPillToggle>
  <ButtonPill
    aria-pressed={false}
    className="md-button-pill-toggle-wrapper"
    data-selected={false}
    disabled={false}
    ghost={true}
    onPress={[Function]}
    size={40}
  >
    <ButtonSimple
      aria-disabled={false}
      aria-pressed={false}
      className="md-button-pill-wrapper md-button-pill-toggle-wrapper"
      data-color="primary"
      data-disabled={false}
      data-ghost={true}
      data-grown={false}
      data-inverted={false}
      data-only-triggers-tooltip={false}
      data-outline={false}
      data-selected={false}
      data-shallow-disabled={false}
      data-size={40}
      isDisabled={false}
      onPress={[Function]}
    >
      <FocusRing
        disabled={false}
      >
        <FocusRing
          disabled={false}
          focusClass="md-focus-ring-wrapper children"
          focusRingClass="md-focus-ring-wrapper children"
        >
          <button
            aria-pressed={false}
            className="md-button-pill-wrapper md-button-pill-toggle-wrapper md-button-simple-wrapper"
            data-color="primary"
            data-disabled={false}
            data-ghost={true}
            data-grown={false}
            data-inverted={false}
            data-only-triggers-tooltip={false}
            data-outline={false}
            data-selected={false}
            data-shallow-disabled={false}
            data-size={40}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            type="button"
          />
        </FocusRing>
      </FocusRing>
    </ButtonSimple>
  </ButtonPill>
</ButtonPillToggle>
`;

exports[`<ButtonPillToggle /> snapshot should match snapshot with ariaStateKey being 'aria-expanded' and isSelected being false 1`] = `
<ButtonPillToggle
  ariaStateKey="aria-expanded"
  isSelected={false}
>
  <ButtonPill
    aria-expanded={false}
    className="md-button-pill-toggle-wrapper"
    data-selected={false}
    disabled={false}
    ghost={true}
    isSelected={false}
    onPress={[Function]}
    size={40}
  >
    <ButtonSimple
      aria-disabled={false}
      aria-expanded={false}
      className="md-button-pill-wrapper md-button-pill-toggle-wrapper"
      data-color="primary"
      data-disabled={false}
      data-ghost={true}
      data-grown={false}
      data-inverted={false}
      data-only-triggers-tooltip={false}
      data-outline={false}
      data-selected={false}
      data-shallow-disabled={false}
      data-size={40}
      isDisabled={false}
      isSelected={false}
      onPress={[Function]}
    >
      <FocusRing
        disabled={false}
      >
        <FocusRing
          disabled={false}
          focusClass="md-focus-ring-wrapper children"
          focusRingClass="md-focus-ring-wrapper children"
        >
          <button
            aria-expanded={false}
            className="md-button-pill-wrapper md-button-pill-toggle-wrapper md-button-simple-wrapper"
            data-color="primary"
            data-disabled={false}
            data-ghost={true}
            data-grown={false}
            data-inverted={false}
            data-only-triggers-tooltip={false}
            data-outline={false}
            data-selected={false}
            data-shallow-disabled={false}
            data-size={40}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            type="button"
          />
        </FocusRing>
      </FocusRing>
    </ButtonSimple>
  </ButtonPill>
</ButtonPillToggle>
`;

exports[`<ButtonPillToggle /> snapshot should match snapshot with ariaStateKey being 'aria-expanded' and isSelected being true 1`] = `
<ButtonPillToggle
  ariaStateKey="aria-expanded"
  isSelected={true}
>
  <ButtonPill
    aria-expanded={true}
    className="md-button-pill-toggle-wrapper"
    data-selected={true}
    disabled={false}
    ghost={true}
    isSelected={true}
    onPress={[Function]}
    size={40}
  >
    <ButtonSimple
      aria-disabled={false}
      aria-expanded={true}
      className="md-button-pill-wrapper md-button-pill-toggle-wrapper"
      data-color="primary"
      data-disabled={false}
      data-ghost={true}
      data-grown={false}
      data-inverted={false}
      data-only-triggers-tooltip={false}
      data-outline={false}
      data-selected={true}
      data-shallow-disabled={false}
      data-size={40}
      isDisabled={false}
      isSelected={true}
      onPress={[Function]}
    >
      <FocusRing
        disabled={false}
      >
        <FocusRing
          disabled={false}
          focusClass="md-focus-ring-wrapper children"
          focusRingClass="md-focus-ring-wrapper children"
        >
          <button
            aria-expanded={true}
            className="md-button-pill-wrapper md-button-pill-toggle-wrapper md-button-simple-wrapper"
            data-color="primary"
            data-disabled={false}
            data-ghost={true}
            data-grown={false}
            data-inverted={false}
            data-only-triggers-tooltip={false}
            data-outline={false}
            data-selected={true}
            data-shallow-disabled={false}
            data-size={40}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            type="button"
          />
        </FocusRing>
      </FocusRing>
    </ButtonSimple>
  </ButtonPill>
</ButtonPillToggle>
`;

exports[`<ButtonPillToggle /> snapshot should match snapshot with ghost being false 1`] = `
<ButtonPillToggle
  ghost={false}
>
  <ButtonPill
    aria-pressed={false}
    className="md-button-pill-toggle-wrapper"
    data-selected={false}
    disabled={false}
    ghost={false}
    onPress={[Function]}
    size={40}
  >
    <ButtonSimple
      aria-disabled={false}
      aria-pressed={false}
      className="md-button-pill-wrapper md-button-pill-toggle-wrapper"
      data-color="primary"
      data-disabled={false}
      data-ghost={false}
      data-grown={false}
      data-inverted={false}
      data-only-triggers-tooltip={false}
      data-outline={false}
      data-selected={false}
      data-shallow-disabled={false}
      data-size={40}
      isDisabled={false}
      onPress={[Function]}
    >
      <FocusRing
        disabled={false}
      >
        <FocusRing
          disabled={false}
          focusClass="md-focus-ring-wrapper children"
          focusRingClass="md-focus-ring-wrapper children"
        >
          <button
            aria-pressed={false}
            className="md-button-pill-wrapper md-button-pill-toggle-wrapper md-button-simple-wrapper"
            data-color="primary"
            data-disabled={false}
            data-ghost={false}
            data-grown={false}
            data-inverted={false}
            data-only-triggers-tooltip={false}
            data-outline={false}
            data-selected={false}
            data-shallow-disabled={false}
            data-size={40}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            type="button"
          />
        </FocusRing>
      </FocusRing>
    </ButtonSimple>
  </ButtonPill>
</ButtonPillToggle>
`;

exports[`<ButtonPillToggle /> snapshot should match snapshot with ghost being true 1`] = `
<ButtonPillToggle
  ghost={true}
>
  <ButtonPill
    aria-pressed={false}
    className="md-button-pill-toggle-wrapper"
    data-selected={false}
    disabled={false}
    ghost={true}
    onPress={[Function]}
    size={40}
  >
    <ButtonSimple
      aria-disabled={false}
      aria-pressed={false}
      className="md-button-pill-wrapper md-button-pill-toggle-wrapper"
      data-color="primary"
      data-disabled={false}
      data-ghost={true}
      data-grown={false}
      data-inverted={false}
      data-only-triggers-tooltip={false}
      data-outline={false}
      data-selected={false}
      data-shallow-disabled={false}
      data-size={40}
      isDisabled={false}
      onPress={[Function]}
    >
      <FocusRing
        disabled={false}
      >
        <FocusRing
          disabled={false}
          focusClass="md-focus-ring-wrapper children"
          focusRingClass="md-focus-ring-wrapper children"
        >
          <button
            aria-pressed={false}
            className="md-button-pill-wrapper md-button-pill-toggle-wrapper md-button-simple-wrapper"
            data-color="primary"
            data-disabled={false}
            data-ghost={true}
            data-grown={false}
            data-inverted={false}
            data-only-triggers-tooltip={false}
            data-outline={false}
            data-selected={false}
            data-shallow-disabled={false}
            data-size={40}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            type="button"
          />
        </FocusRing>
      </FocusRing>
    </ButtonSimple>
  </ButtonPill>
</ButtonPillToggle>
`;

exports[`<ButtonPillToggle /> snapshot should match snapshot with isSelected being false 1`] = `
<ButtonPillToggle
  isSelected={false}
>
  <ButtonPill
    aria-pressed={false}
    className="md-button-pill-toggle-wrapper"
    data-selected={false}
    disabled={false}
    ghost={true}
    isSelected={false}
    onPress={[Function]}
    size={40}
  >
    <ButtonSimple
      aria-disabled={false}
      aria-pressed={false}
      className="md-button-pill-wrapper md-button-pill-toggle-wrapper"
      data-color="primary"
      data-disabled={false}
      data-ghost={true}
      data-grown={false}
      data-inverted={false}
      data-only-triggers-tooltip={false}
      data-outline={false}
      data-selected={false}
      data-shallow-disabled={false}
      data-size={40}
      isDisabled={false}
      isSelected={false}
      onPress={[Function]}
    >
      <FocusRing
        disabled={false}
      >
        <FocusRing
          disabled={false}
          focusClass="md-focus-ring-wrapper children"
          focusRingClass="md-focus-ring-wrapper children"
        >
          <button
            aria-pressed={false}
            className="md-button-pill-wrapper md-button-pill-toggle-wrapper md-button-simple-wrapper"
            data-color="primary"
            data-disabled={false}
            data-ghost={true}
            data-grown={false}
            data-inverted={false}
            data-only-triggers-tooltip={false}
            data-outline={false}
            data-selected={false}
            data-shallow-disabled={false}
            data-size={40}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            type="button"
          />
        </FocusRing>
      </FocusRing>
    </ButtonSimple>
  </ButtonPill>
</ButtonPillToggle>
`;

exports[`<ButtonPillToggle /> snapshot should match snapshot with isSelected being true 1`] = `
<ButtonPillToggle
  isSelected={true}
>
  <ButtonPill
    aria-pressed={true}
    className="md-button-pill-toggle-wrapper"
    data-selected={true}
    disabled={false}
    ghost={true}
    isSelected={true}
    onPress={[Function]}
    size={40}
  >
    <ButtonSimple
      aria-disabled={false}
      aria-pressed={true}
      className="md-button-pill-wrapper md-button-pill-toggle-wrapper"
      data-color="primary"
      data-disabled={false}
      data-ghost={true}
      data-grown={false}
      data-inverted={false}
      data-only-triggers-tooltip={false}
      data-outline={false}
      data-selected={true}
      data-shallow-disabled={false}
      data-size={40}
      isDisabled={false}
      isSelected={true}
      onPress={[Function]}
    >
      <FocusRing
        disabled={false}
      >
        <FocusRing
          disabled={false}
          focusClass="md-focus-ring-wrapper children"
          focusRingClass="md-focus-ring-wrapper children"
        >
          <button
            aria-pressed={true}
            className="md-button-pill-wrapper md-button-pill-toggle-wrapper md-button-simple-wrapper"
            data-color="primary"
            data-disabled={false}
            data-ghost={true}
            data-grown={false}
            data-inverted={false}
            data-only-triggers-tooltip={false}
            data-outline={false}
            data-selected={true}
            data-shallow-disabled={false}
            data-size={40}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            type="button"
          />
        </FocusRing>
      </FocusRing>
    </ButtonSimple>
  </ButtonPill>
</ButtonPillToggle>
`;

exports[`<ButtonPillToggle /> snapshot should match snapshot with outline being false 1`] = `
<ButtonPillToggle
  outline={false}
>
  <ButtonPill
    aria-pressed={false}
    className="md-button-pill-toggle-wrapper"
    data-selected={false}
    disabled={false}
    ghost={true}
    onPress={[Function]}
    outline={false}
    size={40}
  >
    <ButtonSimple
      aria-disabled={false}
      aria-pressed={false}
      className="md-button-pill-wrapper md-button-pill-toggle-wrapper"
      data-color="primary"
      data-disabled={false}
      data-ghost={true}
      data-grown={false}
      data-inverted={false}
      data-only-triggers-tooltip={false}
      data-outline={false}
      data-selected={false}
      data-shallow-disabled={false}
      data-size={40}
      isDisabled={false}
      onPress={[Function]}
    >
      <FocusRing
        disabled={false}
      >
        <FocusRing
          disabled={false}
          focusClass="md-focus-ring-wrapper children"
          focusRingClass="md-focus-ring-wrapper children"
        >
          <button
            aria-pressed={false}
            className="md-button-pill-wrapper md-button-pill-toggle-wrapper md-button-simple-wrapper"
            data-color="primary"
            data-disabled={false}
            data-ghost={true}
            data-grown={false}
            data-inverted={false}
            data-only-triggers-tooltip={false}
            data-outline={false}
            data-selected={false}
            data-shallow-disabled={false}
            data-size={40}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            type="button"
          />
        </FocusRing>
      </FocusRing>
    </ButtonSimple>
  </ButtonPill>
</ButtonPillToggle>
`;

exports[`<ButtonPillToggle /> snapshot should match snapshot with outline being true 1`] = `
<ButtonPillToggle
  outline={true}
>
  <ButtonPill
    aria-pressed={false}
    className="md-button-pill-toggle-wrapper"
    data-selected={false}
    disabled={false}
    ghost={true}
    onPress={[Function]}
    outline={true}
    size={40}
  >
    <ButtonSimple
      aria-disabled={false}
      aria-pressed={false}
      className="md-button-pill-wrapper md-button-pill-toggle-wrapper"
      data-color="primary"
      data-disabled={false}
      data-ghost={true}
      data-grown={false}
      data-inverted={false}
      data-only-triggers-tooltip={false}
      data-outline={true}
      data-selected={false}
      data-shallow-disabled={false}
      data-size={40}
      isDisabled={false}
      onPress={[Function]}
    >
      <FocusRing
        disabled={false}
      >
        <FocusRing
          disabled={false}
          focusClass="md-focus-ring-wrapper children"
          focusRingClass="md-focus-ring-wrapper children"
        >
          <button
            aria-pressed={false}
            className="md-button-pill-wrapper md-button-pill-toggle-wrapper md-button-simple-wrapper"
            data-color="primary"
            data-disabled={false}
            data-ghost={true}
            data-grown={false}
            data-inverted={false}
            data-only-triggers-tooltip={false}
            data-outline={true}
            data-selected={false}
            data-shallow-disabled={false}
            data-size={40}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            type="button"
          />
        </FocusRing>
      </FocusRing>
    </ButtonSimple>
  </ButtonPill>
</ButtonPillToggle>
`;
